backend/de.metas.contracts/src/main/java/de/metas/contracts/commission/commissioninstance/services/repos/CommissionInstanceRepository.java

```diff
@@ -380,11 +385,24 @@ private ImmutableMap<CommissionShare, I_C_Commission_Share> syncShareRecords(
 			result.put(share, shareRecord);
 		}
 
-		shareRecordsToDelete.forEach(InterfaceWrapperHelper::delete);
+		deleteShareRecords(shareRecordsToDelete);
```

Comment:
i think it would be better to add a "before-delete" method to `de.metas.contracts.commission.commissioninstance.interceptor.C_Commission_Share`.
The hardcoded SQL is faster, but i think too dirty.

also, 
- please log to Loggable when a share is deleted (with commission-instance, sales rep, level and contract )
- replace the lambda with an old-school-for